packages/framework/ai-collab/src/explicit-strategy/agentEditReducer.ts

```diff
@@ -552,6 +596,216 @@ function assertObjectIdsExist(treeEdit: TreeEdit, idGenerator: IdGenerator): voi
 	}
 }
 
+const createNodePathRecursive = (
+	node: TreeNode | undefined,
+	idGenerator: IdGenerator,
+	currentPath: NodePath,
+): NodePath => {
+	if (node === undefined) {
+		return currentPath;
+	}
+
+	currentPath.push({
+		shortId: Tree.shortId(node),
+		schemaIdentifier: Tree.schema(node).identifier,
+		parentField: Tree.key(node),
+	});
+
+	const parentNode = Tree.parent(node);
+	return createNodePathRecursive(parentNode, idGenerator, currentPath);
+};
+
```

Comment:
I see a couple of comments below indicating that the diff creation methods need to be called at specific times (before or after) application of the edit. I'm curious why these restrictions are needed. We should be sure to document any such restrictions on each function as applicable (in case any of the other creation functions need these docs).

Reply:
This function is only called once in "insert" case block. The reason why it make the insert diff after insert node succeeded because the index of node that might be needed to undo this operation. Same reason for move that diff made after move node(s) succeeded because we need the original index to put back the node(s).

Reply:
Let's capture that in the docs :)

Reply:
Update comments